Experiment automation/(LUMICKS) Reference kit automation script/Catch_beads_DNA_record_Kymo_for_LacI_version_1.py

```diff
@@ -0,0 +1,318 @@
+        bl.mirror1.move_by(dx=0.1 if dx > 0 else -0.1, speed=speed)
+        dx = target - distance.latest_value
+        throw_if_beads_lost(match_threshold)
+
+
+def goto_force(target, match_threshold, speed=1, tolerance=1, tether_lost_threshold=5):
+    """Move trap 1 until it reaches the `target` force on trap 2.
+
+    Note: This throws an error if the beads are lost"""
+    df = target - force.latest_value
+    throw_if_beads_lost(match_threshold)
+
+    while abs(df) > tolerance:  # um
+        bl.mirror1.move_by(dx=0.05 if df > 0 else -0.05, speed=speed)
+        df = target - get_force(dt=0.2)
+        throw_if_beads_lost(match_threshold)
+    check_tether_breakage(tether_lost_threshold)
+
+
+def get_force(dt=0.5):
+    t0 = bl.timeline.now
+    bl.pause(dt)
+    t1 = bl.timeline.now
+    f = np.mean(force[t0:t1].data)
+
+    return f
+
+
+def catch_dna(min_distance, max_distance, match_threshold, force_threshold, fishing_speed, fishing_attempts, dt):
+    """Moves to the DNA channel and starts oscillating the trap until a prescribed force threshold is reached."""
+    print("Moving to DNA channel")
+    bl.microstage.move_to(name_dna_channel)
+    bl.pause(2.0)
+    print("reset force")
+    bl.reset_force()
+    f_current = get_force(dt=dt)
+
+    attempts = 0
+    max_retries = fishing_attempts*fishing_attempts
+    while f_current < force_threshold:
+        if attempts > max_retries:
+            raise RuntimeError(f"Max DNA fishing attempts {max_retries} reached.")
+
+        if attempts % fishing_attempts == 0:  # If the number of attempts is an integer times max_retries, do this:
+            print("Moving to DNA channel")
+            bl.microstage.move_to(name_dna_channel)
+            print(f"Moving beads a distance {dna_length} um apart")
+            goto_distance(dna_length, match_threshold, speed=fishing_speed)
+            bl.pause(3.0)
+            print("Moving to dna channel")
+            bl.microstage.move_to(name_dna_channel)
+            bl.pause(1.0)
+        print(f"Fishing for DNA: attempt {attempts}/{max_retries}")
+        goto_distance(min_distance, match_threshold, speed=fishing_speed)
+        bl.pause(0.5)
+        goto_distance(max_distance, match_threshold, speed=fishing_speed)
+        f_current = get_force(dt=dt)
+
+        attempts += 1
+
+
+def make_kymograph(force_kymo, match_threshold, name, path):
+    goto_force(force_kymo, match_threshold)
+    try:
+        goto_force(force_kymo, match_threshold)
+        bl.timeline.mark_begin(name)
+        bl.confocal.start_scan()
+        print("start kymograph")
+        bl.pause(10.0)
+        # for df in [10]:
+        print(f"go to a force of {force_kymo} pN")
+        goto_force(force_kymo, match_threshold, speed=1)
+        bl.pause(30)       ## needs to be 300 s in final version
+    finally:
+        print("Stop kymograph")
+        bl.confocal.abort_scan()
+        bl.timeline.mark_end(export=True if path else False, filepath=f"{path}/{name}.h5")

```

Comment:
Would suggest using `pathlib` as it is actually aware of paths.
```suggestion
        bl.timeline.mark_end(export=True if path else False, filepath=str(path / f"{name}.h5"))
```
Not a 100% sure whether the `str` is needed here. Something that would be nice to test.

Reply:
the current version of the script works. Does that mean I should not include this update?